### PR TITLE
Migrate plugin type settings to plugin rez configs

### PR DESCRIPTION
**Pull request by BryceGattis**
Migrate plugin type settings to plugin rez configs so we can more easily add plugin settings to our documentation.

This change is based on JC's suggestion [here](https://github.com/AcademySoftwareFoundation/rez/pull/1689#issuecomment-2002575684).

### CODE REVIEW

File: src/rez/rezconfig.py [hunk 1]
```diff
@@ -1212,46 +1212,7 @@
 # common to all plugins of that type.
 
 plugins = {
```
Comment (JeanChristopheMorinPerso) [inline src/rez/rezconfig.py:1214]: I'm curious to know what would happen if we removed this entirely from here.

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
Tests should pass if you rebase your branch.

File: src/rez/rezconfig.py (hunk 1 above)
Comment (BryceGattis) [reply src/rez/rezconfig.py:1214]: Running `rez config plugins` is how I've been verifying that my changes are working. Removing this line and attempting to run `rez config plugins` causes an error:

```
Traceback (most recent call last):
  File "e:\code\rez\venv\Lib\site-packages\rez\cli\config.py", line 53, in command
    data = data[key]
           ~~~~^^^^^
KeyError: 'plugins'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "E:\Code\rez\venv\Scripts\rez\rez.exe\__main__.py", line 7, in <module>
  File "e:\code\rez\venv\Lib\site-packages\rez\cli\_entry_points.py", line 69, in run_rez
    return run()
           ^^^^^
  File "e:\code\rez\venv\Lib\site-packages\rez\cli\_main.py", line 189, in run
    returncode = run_cmd()
                 ^^^^^^^^^
  File "e:\code\rez\venv\Lib\site-packages\rez\cli\_main.py", line 181, in run_cmd
    return func(opts, opts.parser, extra_arg_groups)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "e:\code\rez\venv\Lib\site-packages\rez\cli\config.py", line 55, in command
    raise ValueError("no such setting: %r" % opts.FIELD)
ValueError: no such setting: 'plugins'
```

**Comment by JeanChristopheMorinPerso** [comment]
EasyCLA check is passing now.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
Thanks!

**Pull request merged**